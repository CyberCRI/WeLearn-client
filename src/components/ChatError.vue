<script setup lang="ts">
import ButtonComponent from './ButtonComponent.vue';

const reload = () => {
  window.location.reload();
};
</script>

<template>
  <div class="chat-error">
    <div>Oops something wrong hapened. Please try again later.</div>
    <ButtonComponent color="secondary" @click="reload()">
      {{ $t('reload') }}
    </ButtonComponent>
  </div>
</template>

<style scoped>
.chat-error {
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 100%;
  text-align: center;
  font-size: 1.125em;
  color: var(--neutral-20);
}
</style>
