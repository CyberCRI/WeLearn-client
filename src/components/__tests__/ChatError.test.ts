import { describe, it, expect } from 'vitest';

import { mount } from '@vue/test-utils';
import ChatError from '../ChatError.vue';

describe('ChatError', () => {
  it('renders properly', () => {
    const wrapper = mount(ChatError);
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.text()).toContain('Oops something wrong hapened. Please try again later.');
  });
});
