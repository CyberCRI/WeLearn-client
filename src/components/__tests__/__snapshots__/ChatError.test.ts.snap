// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChatError > renders properly 1`] = `
<div
  class="chat-error"
  data-v-9a0d2e9c=""
>
  <div
    data-v-9a0d2e9c=""
  >
    Oops something wrong hapened. Please try again later.
  </div>
  <button
    class="button is-secondary"
    data-v-9a0d2e9c=""
  >
    
    reload
    
  </button>
</div>
`;
